File: gachisa-backend/src/main/java/com/gachisa/queue/service/QueueAdmissionScheduler.java
package com.gachisa.queue.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class QueueAdmissionScheduler {

    private final QueueService queueService;

    @Scheduled(fixedDelayString = "${queue.admission-scan-delay-ms:1000}")
    public void processQueues() {
        try {
            queueService.processAllQueues();
        } catch (RuntimeException exception) {
            log.warn("대기열 처리 중 오류가 발생했습니다.", exception);
        }
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/queue/service/QueueAdmissionScheduler.kt
package com.gachisa.queue.service

import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class QueueAdmissionScheduler(private val queueService: QueueService) {
    private val log = LoggerFactory.getLogger(QueueAdmissionScheduler::class.java)

    @Scheduled(fixedDelayString = "\${queue.admission-scan-delay-ms:1000}")
    fun processQueues() {
        try {
            queueService.processAllQueues()
        } catch (exception: RuntimeException) {
            log.warn("대기열 처리 중 오류가 발생했습니다.", exception)
        }
    }
}
